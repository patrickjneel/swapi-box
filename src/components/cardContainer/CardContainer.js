import React from 'react';
import Card from '../card/Card';
import './cardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({itemData, location, updateFavorite, favorites}) => {
  console.log(itemData)
  if(location === 'home') {
    return (
      <img className="at-at" src={require("../../at-at.gif")} />
    )
  } else if(!itemData.length) {
    return (
      <img className="droid" src={require("../../tiefighter.gif")} />
    )
  }
  else {
  const persons = itemData.map(card => {
    let favoritedCard = favorites.find(favorite => favorite.name === card.name);
    let imperialClass = favoritedCard ? 'imperial favorited' : 'imperial';  
    return <Card
      itemData={card}
      key={card.name}
      updateFavorite={updateFavorite}
      location={location}
      type={imperialClass}
    />;
  });
  
  return (
    <div className="container">
      {persons}
    </div>
  );
};
}
export default CardContainer;

CardContainer.propTypes = {
  itemData: PropTypes.array,
  location: PropTypes.string,
  updateFavorite: PropTypes.func,
  favorites: PropTypes.array

};
