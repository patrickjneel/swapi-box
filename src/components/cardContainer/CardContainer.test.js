import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer Test', () => {

  it('should receive props', () => {
    const mockFunc = jest.fn();
    const mockData = [
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
      },
      {
        homeworld: "Tatooine", 
        name: "Luke SkyWalker", 
        species: "Human", 
        population: "200000"
      }
    ];
    const mockLocation = 'people';
    const mockFavs = [];
    const renderedContainer = mount(
      <CardContainer 
        itemData={mockData} 
        location={mockLocation} 
        updateFavorite={mockFunc} 
        favorites={mockFavs}
      />);

    expect(renderedContainer.props().itemData).toEqual(mockData);
    expect(renderedContainer.props().location).toEqual(mockLocation);
    expect(renderedContainer.props().updateFavorite).toEqual(mockFunc);
    expect(renderedContainer.props().favorites).toEqual(mockFavs);

  });

  it.skip('should match snapShot of CardContainer', () => {
    const mockFunc = jest.fn();
    const mockData = 
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
      };
    
    const mockLocation = 'people';
    const mockFavs = [
      {
        homeworld: "Tatooine", 
        name: "Darth Vader", 
        species: "Human", 
        population: "200000"
      }

    ];
    const renderedContainer = shallow(
      <CardContainer 
        itemData={mockData} 
        location={mockLocation} 
        updateFavorite={mockFunc} 
        favorites={mockFavs}
      />);

    expect(renderedContainer).toMatchSnapshot();
  });
});
