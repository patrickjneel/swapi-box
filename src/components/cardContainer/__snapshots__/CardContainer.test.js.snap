// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Test should match snapShot of CardContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "name": "Darth Vader",
              "population": "200000",
              "species": "Human",
            },
          ]
    }
    itemData={
        Array [
            Object {
              "homeworld": "Tatooine",
              "name": "Darth Vader",
              "population": "200000",
              "species": "Human",
            },
          ]
    }
    location="people"
    updateFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          itemData={
                    Object {
                              "homeworld": "Tatooine",
                              "name": "Darth Vader",
                              "population": "200000",
                              "species": "Human",
                            }
          }
          location="people"
          type="imperial favorited"
          updateFavorite={[Function]}
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Darth Vader",
        "nodeType": "function",
        "props": Object {
          "itemData": Object {
            "homeworld": "Tatooine",
            "name": "Darth Vader",
            "population": "200000",
            "species": "Human",
          },
          "location": "people",
          "type": "imperial favorited",
          "updateFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            itemData={
                        Object {
                                    "homeworld": "Tatooine",
                                    "name": "Darth Vader",
                                    "population": "200000",
                                    "species": "Human",
                                  }
            }
            location="people"
            type="imperial favorited"
            updateFavorite={[Function]}
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Darth Vader",
          "nodeType": "function",
          "props": Object {
            "itemData": Object {
              "homeworld": "Tatooine",
              "name": "Darth Vader",
              "population": "200000",
              "species": "Human",
            },
            "location": "people",
            "type": "imperial favorited",
            "updateFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
